backend/app/routes/web/search_suggestions.py

```diff
@@ -73,22 +73,28 @@ async def _search(
         3, description="Row count per collection", gte=3, lt=10
     ),
     search=Depends(search_dep),
-) -> Tuple[str, Dict]:
+) -> tuple[Any, Any] | tuple[str | Any, list[Any]]:
     """Performs the search in a single collection"""
     if "provider" in collection:
         qf = PROVIDER_QF
-    async with AsyncClient() as client:
-        response = await search(
-            client,
-            collection,
-            q=q,
-            qf=qf,
-            fq=fq,
-            sort=DEFAULT_SORT,
-            rows=results_per_collection,
-            exact=exact,
-        )
 
-    res_json = response.data
-    collection = response.collection
-    return collection, res_json["response"]["docs"]
+    try:
+        async with AsyncClient() as client:
+            response = await search(
+                client,
+                collection,
+                q=q,
+                qf=qf,
+                fq=fq,
+                sort=DEFAULT_SORT,
+                rows=results_per_collection,
+                exact=exact,
+            )
+
+        res_json = response.data
+        collection = response.collection
+        return collection, res_json["response"]["docs"]
+
+    except Exception as e:  # pylint: disable=broad-exception-caught
```

Comment:
Too broad. 
we can exactly catch the precise exception that we are expecting https://github.com/cyfronet-fid/eosc-search-service/blob/de4b6be5291d2c14f06b62329916316cda4a9169/backend/app/solr/error_handling.py#L30